backend/cfg/cfg_reducibility.ml

```diff
@@ -0,0 +1,84 @@
+[@@@ocaml.warning "+a-30-40-41-42"]
+
+open! Int_replace_polymorphic_compare [@@ocaml.warning "-66"]
+
+let fatal = Misc.fatal_errorf
+
+let rec has_cycle : Label.Set.t -> Label.Set.t Label.Tbl.t -> bool =
```

Comment:
This algorithm traverses all blocks many times, which can get  very expensive on some awkward graphs. Why did you choose this approach instead of a recursive graph traversal?

Reply:
I thought I was more likely to get this
one right, but indeed switching to
DFS is an option. I think both have the
same complexity, but the current
implementation is likely performing
more allocations.